app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,81 @@
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Class for providing jwt identity from the decoded token from the field specified in a secret
+      class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+        def initialize(
+          authentication_parameters,
+          fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+          resource_class: ::Resource
+        )
+          @logger = Rails.logger
```

Comment:
this is a dependency and should move to the initialzier signatue

Reply:
Done